addOns/exim/src/main/java/org/zaproxy/addon/exim/har/HarImporter.java

```diff
@@ -118,29 +143,128 @@ private static void setHttpResponse(HarResponse harResponse, HttpMessage message
         }
     }
 
-    private void importHarFile(File file) {
+    private void importHarFile(HarLog log) {
         try {
-            processMessages(file);
+            processMessages(log);
             Stats.incCounter(ExtensionExim.STATS_PREFIX + STATS_HAR_FILE);
             success = true;
         } catch (IOException e) {
             LOGGER.warn(
-                    Constant.messages.getString(
-                            ExtensionExim.EXIM_OUTPUT_ERROR, file.getAbsolutePath()));
+                    Constant.messages.getString(ExtensionExim.EXIM_OUTPUT_ERROR, e.getMessage()));
             LOGGER.warn(e);
             Stats.incCounter(ExtensionExim.STATS_PREFIX + STATS_HAR_FILE_ERROR);
             success = false;
         }
     }
 
-    private void processMessages(File file) throws IOException {
-        List<HttpMessage> messages =
-                HarImporter.getHttpMessages(new HarFileReader().readHarFile(file));
-        int count = 1;
+    private HarLog preProcessHarLog(HarLog log) {
```

Comment:
Would be great to have tests that cover these preprocessing checks.

Reply:
I think I mentioned this in an email or the Friday notes at some point. I'd like to add all the HAR files we've collected along the way and use those for tests if there's no objection. None of them are terribly huge, and it's just text so it compresses well (for git operations etc).

Reply:
We should still have specific tests for these, as it's easier to see what input tests what.

Reply:
Do both?

Reply:
I'm fine with both.

Reply:
Better to verify the messages are imported with the expected content (that's the most important).

Reply:
That's fair, just gotta get the tests behaving well first.